src/modules/flow/iio/nodes.c

```diff
@@ -71,33 +102,47 @@ reader_cb(void *data, struct sol_iio_device *device)
 }
 
 static bool
-create_device_channels(struct gyroscope_data *mdata, int device_id)
+create_device_channels(struct iiodevice_data *mdata, int device_id)
 {
     struct sol_iio_channel_config channel_config = SOL_IIO_CHANNEL_CONFIG_INIT;
 
     mdata->device = sol_iio_open(device_id, &mdata->config);
     SOL_NULL_CHECK(mdata->device, false);
 
-#define ADD_CHANNEL(_axis) \
-    if (!mdata->use_device_default_scale) \
-        channel_config.scale = mdata->scale._axis; \
-    if (!mdata->use_device_default_offset) \
-        channel_config.offset = mdata->offset._axis; \
-    mdata->channel_ ## _axis = sol_iio_add_channel(mdata->device, "in_anglvel_" # _axis, &channel_config); \
-    SOL_NULL_CHECK_GOTO(mdata->channel_ ## _axis, error);
+    if (!mdata->use_device_default_scale)
+        channel_config.scale = mdata->scale.x;
 
-    ADD_CHANNEL(x);
-    ADD_CHANNEL(y);
-    ADD_CHANNEL(z);
+    if (!mdata->use_device_default_offset)
+        channel_config.offset = mdata->offset.x;
+    
+    mdata->channels[0] = sol_iio_add_channel(mdata->device, "in_anglvel_x", &channel_config);
+    SOL_NULL_CHECK_GOTO(mdata->channels[0], error);
+
+    if (!mdata->use_device_default_scale)
+        channel_config.scale = mdata->scale.y;
+
+    if (!mdata->use_device_default_offset)
+        channel_config.offset = mdata->offset.y;
+
+    mdata->channels[1] = sol_iio_add_channel(mdata->device, "in_anglvel_y", &channel_config);
+    SOL_NULL_CHECK_GOTO(mdata->channels[1], error);    
+
+    if (!mdata->use_device_default_scale)
+        channel_config.scale = mdata->scale.z;
+
+    if (!mdata->use_device_default_offset)
+        channel_config.offset = mdata->offset.z;
+
+    mdata->channels[2] = sol_iio_add_channel(mdata->device, "in_anglvel_z", &channel_config);
+    SOL_NULL_CHECK_GOTO(mdata->channels[2], error);
 
-#undef ADD_CHANNEL
 
     sol_iio_device_start_buffer(mdata->device);
 
     return true;
 
 error:
-    SOL_WRN("Could not create iio/gyroscope node. Failed to open IIO device %d",
```

Comment:
It would still be interesting to know which type of node failed. I think you can get this information via `enum iiodevice_types type`.